pkg/llmstrategy/chatbased/strategy.go

```diff
@@ -39,84 +41,116 @@ type Strategy struct {
 
 	// PromptTemplateFile allows specifying a custom template file
 	PromptTemplateFile string
-	PreviousQueries    []string
 
 	// Recorder captures events for diagnostics
 	Recorder journal.Recorder
 
 	RemoveWorkDir bool
 
-	MaxIterations    int
-	CurrentIteration int
+	MaxIterations int
 
 	Kubeconfig          string
 	AsksForConfirmation bool
 
 	Tools tools.Tools
 }
 
-// ExecuteChatBased executes a chat-based agentic loop with the LLM using function calling.
-func (a *Strategy) RunOnce(ctx context.Context, query string, previousQueries []string, u ui.UI) error {
+type Conversation struct {
+	strategy *Strategy
```

Comment:
this name is somehow not that intuitive but we will live with it for now.

In my mind, I think of this as a the agentic loop, so sort of a `looper` or `runner`. May be we can rename this to `AgentRunner` or something.

Reply:
To clarify (because I got confused), you mean a better name than Strategy? (vs Conversation)  I agree, Strategy is not a great name ... it is too abstract :-)

Reply:
I was referring to `Strategy` not being a good name.

Reply:
Agreed and have no objection to trying any better names you come up with :-)